src/mongosh/schemas/curriculum-2.0.0.json

```diff
@@ -0,0 +1,103 @@
+{
+	"description": "A Learning resource or other activity that is on a person's career or learning roadmap",
+	"bsonType": "object",
+	"required": [
+		"_id",
+		"resources"
+	],
+	"properties": {
+		"_id": {
+			"description": "The unique identifier for a resource used by a person",
+			"bsonType": "objectId"
+		},
+		"name": {
+			"description": "Temporary property for future removal",
+			"bsonType": "string"
+		},
+		"version": {
+			"description": "Schema Version (only on Version document)",
+			"bsonType": "string",
+			"maxLength": 20
+		},
+		"resources": {
+			"description": "An array of resources",
+			"bsonType": "array",
+			"items": {
+				"bsonType": "object",
+				"required": ["type", "review", "roadmap"],
+				"properties": {
+					"type": {
+						"description": "Type of the resource (topic or adhoc)",
+						"enum": ["topic", "adhoc"]
+					},
+					"resourceId": {
+						"description": "The ID of the resource assigned",
+						"bsonType": "objectId"
+					},
+					"topicId": {
+						"description": "The ID of the topic associated with the resource",
+						"bsonType": "objectId"
+					},
+					"name": {
+						"description": "The Resource Short Name for adhoc resources",
+						"bsonType": "string",
+						"maxLength": 64
+					},
+					"link": {
+						"description": "The Resource Link for adhoc resources",
+						"bsonType": "string",
+						"maxLength": 64
+					},
+					"review": {
+						"description": "Free text review of the person's experience with this resource",
+						"bsonType": "string",
+						"maxLength": 256
+					},
+					"roadmap": {
+						"description": "The top level grouping Now, Next, Later, Celebrated",
+						"enum": ["completed", "Celebrated", "now", "next", "later"]
+					},
+					"status": {
+						"description": "Archived, Assigned, Started, Completed",
+						"bsonType": "string"
+					},
+					"statusDate": {
+						"description": "Date the status was last changed",
+						"bsonType": "date"
+					},
+					"started": {
+						"description": "Date/time the link was clicked",
+						"bsonType": "date"
+					},
+					"completed": {
+						"description": "Date/time the item was checked off",
+						"bsonType": "date"
+					},
+					"rating": {
+						"description": "Rating 0-5",
+						"bsonType": "int",
+						"minimum": 0,
+						"maximum": 5
+					}
+				},
+				"oneOf": [
+					{
```

Comment:
I like what you're documenting here, but I've not used the oneOf constraint before in a mongosh schema, and I don't think that this structure has any impact unless properties are required. We should not have this many required properties because of the auto save functionality that we're using. You should think about the data as a new created one time. I think the only required field we should have this entire document is _ID